build/references/coreth-arc20s.md

```diff
@@ -0,0 +1,108 @@
+
+#### EVM Transaction
+
+An EVM Transaction is composed of the following fields:
+
+* **`nonce`** Scalar value equal to the number of transactions send by the sender
+* **`gasPrice`** Scalar value equal to tthe number of Wei paid per unit of gas to execute this transaction
+* **`gasLimit`** Scalar value equal to the maximum amount of gas that should be used in executing this transaction
+* **`to`** The 20 byte address of the message call's recipient. If the transaction is creating a contract, `to` is left empty
+* **`value`** Scalar value of native asset (AVAX) measured in wei to be transferred to the message call's recipient or in the case of a contract creation, as an endowment to the newly created contract.
+* **`v, r, s`** Values corresponding to the signature of the transaction.
+* **`data`** Unlimited size byte array specifying the input data to a contract call or (if creating a contract) the EVM bytecode for the account initialisation process
+
+#### Native Assets
+
+Non-AVAX Assets, however, have no counterpart within the EVM. Therefore, Coreth makes minor modifications to the EVM in order to support asset balances and the ability to transfer these assets on the C Chain.
+
+Non-AVAX Assets (ANTs) are supported on the C Chain by keeping an account balance mapping [assetID -> balance]. These assets can be exported back to the X Chain or moved around on the C Chain using `assetCall` and `assetBalance`.
+
+##### assetCall
+
+`assetCall` is a precompiled contract at address `0xassetCallAddr`. `assetCall` allows users to atomically transfer a native asset to a given address and also make a contract call to that same address.
+
+This is identical to how a normal transaction can send value to some address, while also atomically calling that address with some `data`.
+
+
+```text
+assetCall(address addr, uint256 assetID, uint256 assetAmount, bytes memory callData) -> {ret: bytes memory}
+```
+
+These arguments can be packed by `abi.encodePacked(...)` in Solidity since there is only one argument with variadic length (callData). The first three arguments are constant length, so the precompiled contract simply parses the call `data` input as:
+
+
+```text
++-------------+---------------+--------------------------------+
+| address     : address       |                       20 bytes |
++-------------+---------------+--------------------------------+
+| assetID     : uint256       |                       32 bytes |
++-------------+---------------+--------------------------------+
+| assetAmount : uint256       |                       32 bytes |
++-------------+---------------+--------------------------------+
+| callData    : bytes memory  |            len(callData) bytes |
++-------------+---------------+--------------------------------+
+                              |       84 * len(callData) bytes |
```

Comment:
This should be `84 + len(callData) bytes` and not `84 * len(callData) bytes`

Replace `*` with `+`

Reply:
Good catch